numba/core/config.py

```diff
@@ -400,9 +400,15 @@ def avx_default():
                 # on some CPUs (list at
                 # http://llvm.org/bugs/buglist.cgi?quicksearch=avx).
                 # For now we'd rather disable it, since it can pessimize code
-                cpu_name = ll.get_host_cpu_name()
-                return cpu_name not in ('corei7-avx', 'core-avx-i',
-                                        'sandybridge', 'ivybridge')
+                cpu_name = CPU_NAME or ll.get_host_cpu_name()
```

Comment:
I guess that this is technically fixing a bug, in that a prescribed `NUMBA_CPU_NAME` was previously ignored in the logic surrounding AVX use. Should a test be added that runs if `get_host_cpu_name()` returns anything other than `nocona` and the test asserts that a prescribed `NUMBA_CPU_NAME=nocona` would result in `config.ENABLE_AVX==False`?

Reply:
Added test in https://github.com/numba/numba/pull/9603/commits/50af0ab8dc3fd5031b6b420b012fe9904b630752